src/qt/bitcoingui.cpp

```diff
@@ -728,6 +735,14 @@ void BitcoinGUI::optionsClicked()
     dlg.exec();
 }
 
+void BitcoinGUI::openConfigClicked()
+{
+    boost::filesystem::path pathConfig = GetConfigFile();
+    /* Open gridcoinresearch.conf with the associated application */
+    if (boost::filesystem::exists(pathConfig))
+        QDesktopServices::openUrl(QUrl::fromLocalFile(QString::fromStdString(pathConfig.string())));
```

Comment:
This doesn't seem to work reliably on Windows: 

> GUI: ShellExecute 'file:///C:/Users/Cy/AppData/Roaming/GridcoinResearch/gridcoinresearch.conf' failed (error 31).

...since `QDesktopServices::openUrl()` appears to wrap the Win32 `ShellExecute` API, and this API needs a program associated with .conf files to choose a handler for the URL.

The button that opens the debug.log file in the Debug Console seems to work without trouble because Windows associates .log files with Notepad by default. We will need to find an alternative solution for the config file.

---

Linux seems fine as long as the `xdg-open` (or similar) program is available. When it isn't, the menu doesn't do anything, and Gridcoin outputs this log message: 

> GUI: Unable to detect a launcher for 'file:///home/cy/.GridcoinResearch/gridcoinresearch.conf'

Although it's harmless, for Linux, we may be able hide the menu item when it will fail by detecting one of the `XDG_*` environment variables to avoid confusion.

---

I will give macOS a try when I can steal the Mac in the house 🙂

Reply:
On that note, opening the data directory may be easier because most desktop environments should associate a handler for a directory with a file manager.

Reply:
@cyrossignol It should now also work under Windows. Bitcoin also doesn't use fallback if xdg-open is not available, what's your opinion on that?

Reply:
We can think of a way to address that later. This is a pretty good start for now.